spring-aot/src/main/java/org/springframework/context/bootstrap/generator/nativex/HintsBeanNativeConfigurationProcessor.java

```diff
@@ -62,6 +64,7 @@
 	@Override
 	public void process(BeanInstanceDescriptor descriptor, NativeConfigurationRegistry registry) {
 		findAndRegisterRelevantNativeHints(descriptor.getUserBeanClass(), registry);
+		registerReflectionForInitializtionMethodsIfNecessary(descriptor, registry);
```

Comment:
This is not the right place for this. This should go in `DefaultBeanNativeConfigurationProcessor`.